lib/elixir/lib/kernel.ex

```diff
@@ -932,6 +937,11 @@ defmodule Kernel do
   Removes the first occurrence of an item on the left list
   for each item on the right.
 
+  The complexity of `a -- b` is proportional to `length(a) * length(b)`,
```

Comment:
I should note that I'm actually in the middle of optimizing MapSet
difference, intersection, and a few other functions in light of the stuff
we learned during my experimentation with maps a while back. I'm seeing
improvements in excess of 2x for certain cases.

How about we revisit this wording after?

On Friday, March 18, 2016, Eric Meadows-Jönsson <notifications@github.com>
wrote:

> In lib/elixir/lib/kernel.ex
> <https://github.com/elixir-lang/elixir/pull/4400#discussion_r56723938>:
>
> > @@ -932,6 +937,11 @@ defmodule Kernel do
> >    Removes the first occurrence of an item on the left list
> >    for each item on the right.
> >
> > +  The complexity of `a -- b` is proportional to `length(a) * length(b)`,
>
> We should note that this is the worst-case complexity. The ideal case is
> linear.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/elixir-lang/elixir/pull/4400/files/aff02ceaf991a965b27444724d9aa760e7d2b113#r56723938>
>